purchase_order_ubl/models/purchase.py

```diff
@@ -85,7 +86,7 @@ def _ubl_add_order_line(
             seller=self.partner_id.commercial_partner_id,
             currency=self.currency_id, price_subtotal=oline.price_subtotal,
             qty_precision=qty_precision, price_precision=price_precision,
-            version=version)
+            uuid=oline.id, version=version)
```

Comment:
Isn't it unwise to use a numeric id that is not 'universally unique'?

Reply:
Thanks for your review @StefanRijnhart 

![Selection_137](https://user-images.githubusercontent.com/1853434/102979677-b0026780-4506-11eb-9116-732a50c3992f.png)

Semantically you're right, but contextually to the exchange from a system A to a system B for the flow, purchase line item here, the UID is unique because of unique id in the table.

You're right it's more a UID than UUID. It's a compromise.
Do you think there is more convenient attribute ?

http://www.datypic.com/sc/ubl22/e-cac_LineItem.html

Reply:
If there is already an `identifier for this line item asigned by the seller` then that's where the line id should go, not into the UUID.

I can imagine applying a mix-in class to add an automatically populated UUID column to Odoo models. Isn't there already a module that provides that?

Reply:
> If there is already an identifier for this line

I carefully check all other attributes on line item and there is no other matching the case. There nearest is uuid.
I propose to add an overridable method which returns this UUID with default value as False, what do you think ?
If someone need for its usage the proposed mixin mechanism, it can be implemented with this method but in this case it should be decoupled of these EDI module (not depends on it).

Reply:
I do not know my way around these specifications. But in the screenshot above, where it says 'SalesOrderID', the comment mentions the identifier for the *line* item. I see references to `cbc:SalesOrderLineID` if I google around. Is there an error in the specs above?

Reply:
Yes that's quite confusing because we know both that SalesOrderID is very different from SalesOrderLineID.
Specifications are not perfect and our projects specially in v8 are not perfect too ;-)